### PR TITLE
Fixture for all entitlements

### PR DESCRIPTION
**Pull request by rui1610**
## Purpose

Add unit tests for various datasource with entitlements

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe: additional unit tests
```

## How to Test

* Test the code

```
go test ./...
```

## What to Check

* All tests should run successfully

## Other Information

Pre-requisite is a properly setup subaccount and directory for the tests.

### CODE REVIEW

File: internal/provider/datasource_directory_user_test.go
Comment (v0lkc) [inline internal/provider/datasource_directory_user_test.go:1]: It looks to me you committed a bit too much here.

File: internal/provider/datasource_globalaccount_user_test.go
Comment (v0lkc) [inline internal/provider/datasource_globalaccount_user_test.go:1]: Same here

File: internal/provider/datasource_subaccount_user_test.go
Comment (v0lkc) [inline internal/provider/datasource_subaccount_user_test.go:1]: Same here

File: go.mod
Comment (v0lkc) [inline go.mod:1]: Kindly ask you to revert this change, plugin docs must not be listed in go.mod. Instead please install this locally on your machine with

`go install github.com/hashicorp/terraform-plugin-docs@latest`

File: go.sum
Comment (v0lkc) [inline go.sum:1]: You forgot one. `go mod tidy` should do the trick and remove this.

Comment (rui1610) [reply go.sum:1]: Thanks and sorry for missing this one. Just fixed it.
